tests/h/views/activity_test.py

```diff
@@ -594,25 +592,40 @@ def test_search_passes_the_display_name_to_the_template_if_the_user_has_one(
 
         assert username == user.display_name
 
-    def test_search_passes_the_annotation_count_to_the_template(self,
-                                                                controller,
-                                                                search,
-                                                                user,
-                                                                user_service):
-        controller.search()
+    def test_search_passes_the_user_annotation_counts_to_the_template(self,
+                                                                      controller,
+                                                                      pyramid_config,
+                                                                      search,
+                                                                      user):
+        pyramid_config.testing_securitypolicy(user.userid)
+
+        stats = controller.search()['stats']
 
-        user_service.annotation_count.assert_called_once_with(user.userid)
+        assert stats['annotation_count'] == 6
 
-    def test_search_does_not_pass_the_annotation_count_to_the_template_when_flag_is_disabled(self,
-                                                                                             controller,
-                                                                                             search,
-                                                                                             user,
-                                                                                             user_service,
-                                                                                             pyramid_request):
+    def test_search_does_not_pass_the_user_annotation_counts_to_the_template_when_flag_is_disabled(self,
+                                                                                                   controller,
+                                                                                                   pyramid_request,
+                                                                                                   search,
+                                                                                                   user):
         pyramid_request.feature.flags['total_shared_annotations'] = False
-        controller.search()
+        result = controller.search()['stats']
+
+        assert result['annotation_count'] is None
+
+    def test_search_does_not_pass_the_user_annotation_counts_to_the_template_when_user_is_different_from_auth_user(self,
```

Comment:
I still think some of these test names are a bit long and unwieldy. Far more readable to just call the test something short and use a docstring to explain the case you're testing here.